test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -256,3 +256,198 @@
+                              [:minute [[#t "2022-10-02 08:30:00" #t "2022-10-02 08:30:59.999" 0  "millisecond under a minute"]
+                                        [#t "2022-10-02 08:30:00" #t "2022-10-02 08:34:00" 4      "minutes"]
+                                        [#t "2022-10-02 08:30:00" #t "2022-10-02T10:30:00" 120    "hours"]
+                                        [#t "2021-10-02 08:30:00" #t "2022-10-02 08:34:00" 525604 "over a year"]]]
+                              [:second [[#t "2022-10-02 08:30:00" #t "2022-10-02 08:30:00.999" 0    "millisecond under a second"]
+                                        [#t "2022-10-02 08:30:00" #t "2022-10-02 08:34:00" 240      "minutes"]
+                                        [#t "2022-10-02 08:30:00" #t "2022-10-02T10:30:00" 7200     "hours"]
+                                        [#t "2021-10-02 08:30:00" #t "2022-10-02 08:34:00" 31536240 "over a year"]]]]
+
+                [x y expected desc] cases]
+          (testing (name unit)
+            (testing desc
+              (is (= [expected (- expected)] (query x y unit))))))))
+    (mt/dataset temporal-dataset
+      (testing "Can compare across dates, datetimes, and with timezones from a table"
+        ;; these particular numbers are not important, just that we can compare between dates, datetimes, etc.
+        (is (= [[428 397 31]]
+               (mt/rows
+                (mt/run-mbql-query datediff-mixed-types
+                  {:fields [[:expression "tz,dt"]
+                            [:expression "tz,d"]
+                            [:expression "d,dt"]]
+                   :filter [:= $index 1]
+                   :expressions
+                   {"tz,dt" [:datetimediff $tz $dt :day]
+                    "tz,d"  [:datetimediff $tz $d :day]
+                    "d,dt"  [:datetimediff $d $dt :day]}}))))))))
+
+(deftest datetimediff-time-zones-test
+  (mt/test-drivers (mt/normal-drivers-with-feature :datetimediff)
+    (mt/dataset sample-dataset
+      (let [diffs (fn [x y]
+                    (let [units [:second :minute :hour :day :week :month :year]]
+                      (->> (mt/run-mbql-query orders
+                             {:limit 1
+                              :expressions (into {} (for [unit units]
+                                                      [(name unit) [:datetimediff x y unit]]))
+                              :fields (into [] (for [unit units]
+                                                 [:expression (name unit)]))})
+                           mt/rows first
+                           (zipmap units))))]
+        (testing "a day"
+          (mt/with-temporary-setting-values [driver/report-timezone "Atlantic/Cape_Verde"] ; UTC-1 all year
+            (is (partial= {:second 86400 :minute 1440 :hour 24 :day 1}
+                          (diffs #t "2022-10-02T01:00:00Z[+01:00]"     ; 2022-10-01T23:00:00Z[-01:00]
```

Comment:
```suggestion
                          (diffs #t "2022-10-02T01:00:00[+01:00]" 
```
I'm a bit confused by the `Z` since it means UTC, but the offset is +01.
Pretty sure we'll omit it before set SQL parameter, but we should remove it in the test.

Reply:
You're right, this format is just an extra verbose way to write a zoned date time. I've shortened them now.